fix: type conversion for primary keys in create/update

qwerdenkerXD opened the pull request:
**IMPORTANT: [This PR](https://www.github.com/Zendro-dev/graphql-server/pull/106) is a requirement for merging this PR**

Changes in this PR:
 - added a type conversion of the primary key to the create/update resolvers

qwerdenkerXD committed:
fix: type conversion for primary keys in create/update

qwerdenkerXD committed:
fix: type conversion for primary keys in create/update

qwerdenkerXD commented:
An integration test is missing, will be added soon

qwerdenkerXD committed:
fix: null for IDs not allowed on validation side

qwerdenkerXD committed:
fix: null for IDs not allowed on validation side

wunderbarr committed:
hotfix: update integration tests

qwerdenkerXD committed:
update: renamed function castIdIntoRespectiveType -> castIdentifierToCorrectType

qwerdenkerXD committed:
update: renamed function castIdIntoRespectiveType -> castIdentifierToCorrectType

qwerdenkerXD committed:
merge after rebase
rebased because tests were updated in 146627a

qwerdenkerXD committed:
update: added integration tests for internalId type casting
therefore added fisher model associated to river

wunderbarr commented on test/integration_test_misc/integration_test_models_instance1/fisher.json:
```diff
@@ -0,0 +1,21 @@
+    "associations": {
+      "fishingRivers": {
+        "type": "many_to_many",
+        "implementation": "sql_cross_table",
```
not sure why you are using cross table. Actually I think we have discarded this part. You can change it to 
"implementation": "foreignkeys"

qwerdenkerXD committed:
reworked tests
prettified code (like using spaces instead of tabs)
use loops for creating and deleting records in test group 'InternalId type Int: CRUD functions for fisher'
use 'foreignkeys' implementation for association between fisher and river

wunderbarr commented on test/mocha_integration.test.js:
```diff
@@ -12,3850 +14,3982 @@ const itHelpers = require("./integration_test_misc/integration_test_helpers");
-    );
-    resBody = JSON.parse(res.body.toString("utf8"));
-    expect(resBody).to.deep.equal({
-      data: {
```
seems like the indent is not consistent. Previously we use indent with 2 characters. Currently the indent has four characters. Maybe adjust the prettier setup?

qwerdenkerXD committed:
set indent size to 2 in mocha_integration.test.js
